File: app/src/main/java/me/tagavari/airmessage/messaging/viewholder/VHMessagePreviewLink.java
package me.tagavari.airmessage.messaging.viewholder;

import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.drawable.RippleDrawable;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.bumptech.glide.load.engine.Resource;
import com.google.android.material.card.MaterialCardView;
import com.google.android.material.shape.MaterialShapeDrawable;
import com.google.android.material.shape.ShapeAppearanceModel;

import me.tagavari.airmessage.R;
import me.tagavari.airmessage.helper.MessageShapeHelper;
import me.tagavari.airmessage.helper.ResourceHelper;

public class VHMessagePreviewLink extends VHMessagePreview {
	public final View viewBorder;
	public final ImageView imageHeader;
	public final TextView labelTitle;
	public final TextView labelDescription;
	public final TextView labelAddress;
	
	public VHMessagePreviewLink(@NonNull View itemView, View viewBorder, ImageView imageHeader, TextView labelTitle, TextView labelDescription, TextView labelAddress) {
		super(itemView);
		this.viewBorder = viewBorder;
		this.imageHeader = imageHeader;
		this.labelTitle = labelTitle;
		this.labelDescription = labelDescription;
		this.labelAddress = labelAddress;
	}
	
	@Override
	public void updateViewEdges(Context context, boolean anchoredBottom, boolean alignToRight) {
		//Creating the message shape
		ShapeAppearanceModel messageAppearance = MessageShapeHelper.createRoundedMessageAppearanceBottom(context.getResources(), anchoredBottom, alignToRight);
		
		//Assigning the border shape
		MaterialShapeDrawable borderDrawable = new MaterialShapeDrawable(messageAppearance);
		borderDrawable.setFillColor(ColorStateList.valueOf(0));
		borderDrawable.setStroke(ResourceHelper.dpToPx(1), context.getColor(R.color.colorDivider));
		viewBorder.setBackground(borderDrawable);
		
		//Updating the card shape
		((MaterialCardView) itemView).setShapeAppearanceModel(messageAppearance);
	}
}

File: app/src/main/java/me/tagavari/airmessage/messaging/viewholder/VHMessagePreviewLink.kt
package me.tagavari.airmessage.messaging.viewholder

import android.content.Context
import android.content.res.ColorStateList
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.google.android.material.card.MaterialCardView
import com.google.android.material.shape.MaterialShapeDrawable
import me.tagavari.airmessage.R
import me.tagavari.airmessage.helper.MessageShapeHelper
import me.tagavari.airmessage.helper.ResourceHelper

class VHMessagePreviewLink(
	itemView: View,
	val viewBorder: View,
	val imageHeader: ImageView,
	val labelTitle: TextView,
	val labelDescription: TextView,
	val labelAddress: TextView
) : VHMessagePreview(itemView) {
	override fun updateViewEdges(context: Context, anchoredBottom: Boolean, alignToRight: Boolean) {
		//Creating the message shape
		val messageAppearance = MessageShapeHelper.createRoundedMessageAppearanceBottom(context.resources, anchoredBottom, alignToRight)
		
		//Assigning the border shape
		val borderDrawable = MaterialShapeDrawable(messageAppearance)
		borderDrawable.fillColor = ColorStateList.valueOf(0)
		borderDrawable.setStroke(ResourceHelper.dpToPx(1f).toFloat(), context.getColor(R.color.colorDivider))
		viewBorder.background = borderDrawable
		
		//Updating the card shape
		(itemView as MaterialCardView).shapeAppearanceModel = messageAppearance
	}
}